### ISSUE TITLE
Mac compile error

### ISSUE THREAD

**Issue by awhillas** [cantordust/cortex#1]
Hey,
Just tried to compile your project and got this:
```
CMake Error at /usr/local/Cellar/cmake/3.8.2/share/cmake/Modules/ExternalProject.cmake:2005 (message):
  No download info given for 'pugixml' and its source directory:

   /Users/alex/dev/cortex/deps/pugixml/scripts

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * URL
   * GIT_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
   * SVN_REVISION
   * DOWNLOAD_COMMAND
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.8.2/share/cmake/Modules/ExternalProject.cmake:2560 (_ep_add_download_command)
  CMakeLists.txt:122 (ExternalProject_Add)


-- Configuring incomplete, errors occurred!
See also "/Users/alex/dev/cortex/CMakeFiles/CMakeOutput.log".
```
Just started to learn C++ so looking forward to reviewing your code. Glad to see your still at it 👍

**Comment by cantordust** [comment]
Hey, thank you for being a guinea pig! :) I'm working on removing dependencies which require compilation. In the current version the configuration is in JSON format, and the dependency is just a header file. I hope that I'll be able to upload it ASAP.

In the meantime, you can do the following:

`cd ~/dev/cortex/deps/pugixml && make clean`

this removes the compiled library. Then you can proceed with compiling cortex.

**Comment by awhillas** [comment]
still no joy
```
make: *** No rule to make target `clean'.  Stop.
```
(compiling with C/C++ is always such a pain ! :D )

**Comment by cantordust** [comment]
Oops, that should have been

`cd **~/dev/cortex/deps/pugixml/*scripts*/ && make clean && rm CMakeCache.txt`

Mea culpa.

I also found another error. Please download the new CMakeLists.txt.

**Comment by cantordust** [comment]
I updated the CMakeLists.txt and the dependencies. Please try fetching again, running build.sh should just work. Let me know if there are any issues.

**Comment by awhillas** [comment]
```
cd ~/dev/cortex/deps/pugixml/scripts/ && make clean && rm CMakeCache.txt
make: *** No rule to make target `clean'.  Stop.
```
There is now _nothing_ in the `~/dev/cortex/deps/pugixml/scripts/` dir...?

I still haven't gotten to the build/deploy part of C++ so i can't help you here yet :(

**Comment by cantordust** [comment]
Yes, there should be just two `tar.gz` archives in the `deps` directory. Just fetch the latest master and run `$ cd ~/dev/cortex/ && ./build.sh`. I have updated the script, so it will extract and compile everything for you.
